python_modules/dagster/dagster/_config/validate.py

```diff
@@ -107,9 +111,7 @@ def _validate_config(context: ValidationContext, config_value: object) -> Evalua
         if not is_config_scalar_valid(context.config_type_snap, config_value):
             return EvaluateValueResult.for_error(create_scalar_error(context, config_value))
         # If user passes an EnvVar or IntEnvVar to a non-structured run config dictionary, throw explicit error
-        if not context.allow_envvar and context.config_type_snap.scalar_kind == ConfigScalarKind.STRING and isinstance(
-            config_value, (EnvVar, IntEnvVar)
-        ):
+        if not context.allow_envvar and isinstance(config_value, (EnvVar, IntEnvVar)):
```

Comment:
can we change  isinstance(config_value, (EnvVar, IntEnvVar) to some method that is future proof to adding new types of env vars?